src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -371,7 +450,17 @@ public Integer invoke(File f, VirtualChannel channel) throws IOException, Interr
         }
 
         @Override public void cleanup(FilePath workspace) throws IOException, InterruptedException {
-            controlDir(workspace).deleteRecursive();
+            for (int i = 0; i < MAX_CLEANUP_TRIES; i++) {
+                try {
+                    controlDir(workspace).deleteRecursive();
+                    break;
+                } catch (IOException e) {
+                    if (e.getMessage().contains("Unable to delete")) {
+                        Thread.sleep(MS_BETWEEN_RETRIES);
+                        continue;
+                    }
+                }
+            }
```

Comment:
I'm a little confused about this. IIUC, you added it to avoid needing `Thread.sleep` in the Windows tests, but was the behavior seen in the test something that was happening in real usage too? I'm not sure if it makes sense to add this behavior for all platforms. Also, doesn't `deleteRecursive` already have internal retries?

If it's between this or having long calls to `Thread.sleep` in the Windows tests, I would go with the calls to `sleep`. Maybe another option could be to tweak [the properties that control the retries in `Util.deleteRecursive`](https://github.com/jenkinsci/jenkins/blob/97d919c971fedf4ea5c23172359fec959c15829a/core/src/main/java/hudson/Util.java#L1752-L1779) in the Windows tests only.

Reply:
So just to clarify, I was never seeing this in real usage. The only time I ever see this issue is in this test code. Of course my real world testing is just simple loops.

Reply:
There, was able to get the tuning to work with the testing.  See 9bd66fb8cff734d41b3959d32cda54b49f96b8a6

Reply:
of course the solution only works on my local machine and not CI...

Reply:
Okay, I think I got something working. No amount of `FlagRule` or `System.setProperty` was working. Instead, I gave up on that and decided to do something like how we check for zombies in `BourneShellScriptTest` by using `ps`. But insted of `ps` I use `tasklist` and check for the existence of the `durable-task-monitor` binary.

see 0dcb76cc80957432b23cca459f3fa5f1052d42c7